build.gradle

```diff
@@ -18,84 +14,47 @@ sourceSets {
                 'src/main/java',
                 'vendor/sqlite/src/java'
         ]
-        resources {
-            srcDir 'src/resources'
-        }
     }
 }
 
+def jacksonDatabindVersion = '2.5.0'
+
 dependencies {
-    compile 'com.squareup.okhttp3:okhttp:3.11.0'
-    compile 'com.fasterxml.jackson.core:jackson-core:2.5.0'
-    compile 'com.fasterxml.jackson.core:jackson-annotations:2.5.0'
-    compile 'com.fasterxml.jackson.core:jackson-databind:2.5.0'
-    compile 'com.github.oxo42:stateless4j:2.4.0'
+    implementation 'com.squareup.okhttp3:okhttp:3.11.0'
+    implementation "com.fasterxml.jackson.core:jackson-core:$jacksonDatabindVersion"
+    implementation "com.fasterxml.jackson.core:jackson-annotations:$jacksonDatabindVersion"
+    implementation "com.fasterxml.jackson.core:jackson-databind:$jacksonDatabindVersion"
+    implementation 'com.github.oxo42:stateless4j:2.4.0'
 }
 
-task createMavenDirectory(type: Exec) {
-
-    ext {
-        uploadUser = System.getenv("MAVEN_UPLOAD_USERNAME") + ":" + System.getenv("MAVEN_UPLOAD_PASSWORD")
-        mkcolPath = System.getenv("MAVEN_UPLOAD_REPO_URL") + "com/couchbase/lite/couchbase-lite-java-core/" + version + "/"
-    }
-    commandLine "curl", "--user", uploadUser, "-X", "MKCOL", mkcolPath
+task sourcesJar(type: Jar, dependsOn: classes) {
+    classifier 'sources'
+    from sourceSets.main.java.srcDirs
 }
 
-// this hack is only needed for apache mod_dav based Maven repo's like file.couchbase.com.  otherwise, skip it
-createMavenDirectory.onlyIf { System.getenv("MAVEN_UPLOAD_REPO_URL").contains("files") }
+publishing {
+    publications {
+        maven(MavenPublication) {
+            groupId = 'com.troii'
+            artifactId = 'couchbase-lite-java-core'
+            version = "1.4.6"
```

Comment:
1.4.5 ?

Reply:
yes, i just had to bump it up to try if it works -> nexus already holds 1.4.5, but will revert it here too